src/app/terminal.js

```diff
@@ -0,0 +1,301 @@
+        document.addEventListener('mouseup', removeGhostbar)
+        document.addEventListener('keydown', cancelGhostbar)
+      }
+    }
+    function cancelGhostbar (event) {
+      if (event.keyCode === 27) {
+        document.body.removeChild(ghostbar)
+        document.removeEventListener('mousemove', moveGhostbar)
+        document.removeEventListener('mouseup', removeGhostbar)
+        document.removeEventListener('keydown', cancelGhostbar)
+      }
+    }
+    function moveGhostbar (event) { // @NOTE HORIZONTAL ghostbar
+      ghostbar.style.top = self._api.getPosition(event) + 'px'
+    }
+    function removeGhostbar (event) {
+      if (self._view.icon.classList.contains('fa-angle-double-up')) {
+        self._view.icon.classList.toggle('fa-angle-double-down')
+        self._view.icon.classList.toggle('fa-angle-double-up')
+      }
+      document.body.removeChild(ghostbar)
+      document.removeEventListener('mousemove', moveGhostbar)
+      document.removeEventListener('mouseup', removeGhostbar)
+      document.removeEventListener('keydown', cancelGhostbar)
+      self.event.trigger('resize', [self._api.getPosition(event)])
+    }
+
+    return self._view.panel
+    function change (event) {
+      if (event.which === 13) {
+        if (event.ctrlKey) { // <ctrl+enter>
+          // console.log('enter')
+          self._view.input.appendChild(document.createElement('br'))
+          self.scroll2bottom()
+        } else {
+          event.preventDefault()
+        }
+      }
+      console.log(event.which, event.keyCode)
+      // console.log('----------------')
+      // console.log(event.key) // Control
+      // console.log('-----')
+      // console.log(event.code) // ControlLeft
+      // console.log('-----------')
+      // console.log('event.altKey ' + event.altKey)
+      // console.log('event.ctrlKey ' + event.ctrlKey)
+      // console.log('event.metaKey ' + event.metaKey)
+      // console.log('event.shiftKey ' + event.shiftKey)
+      // console.log(event.location)
+      // console.log('event.location')
+      // console.log(event.returnValue)
+      // console.log('event.returnValue')
+      // console.log(event.type)
+      // console.log('event.type')
+      // console.log('-----')
+      // console.log(event)
+    }
+  }
+  log (obj) {
+    var self = this
+    // var entries = self._serialize(obj)
+    // @TODO: convert types to \n string arrays
+
+    var entries = obj.split('\n')
+    self.data.session.push(entries)
+    var block = yo`
+      <div class=${css.block}>
+        ${entries.map(str => document.createTextNode(`${str}\n`))}
+      </div>
+    `
+    self._view.log.appendChild(block)
+    self.scroll2bottom()
+    return entries
+  }
+  scroll2bottom () {
+    var self = this
+    setTimeout(function () {
+      self._view.term.scrollTop = self._view.term.scrollHeight
+    }, 0)
+  }
+  _serialize (obj) { return JSON.stringify(obj, null, 2) }
+  _shell (text) { // default shell
```

Comment:
oh no, this is basically for interpreting the input a user writes into the terminal.
It's not yet added.
Yes, it will "know" about the `web3` object